File: accepted/299-bulls-and-cows.py
# https://leetcode.com/problems/bulls-and-cows/

class Solution(object):
    def getHint(self, secret, guess):
        bulls, cows = 0, 0
        map = [0] * 10
        
        for i in xrange(len(secret)):
            if secret[i] == guess[i]:
                bulls += 1
            else:
                map[int(secret[i])] += 1
            
        for i in xrange(len(secret)):
            if secret[i] != guess[i] and map[int(guess[i])] > 0:
                map[int(guess[i])] -= 1
                cows += 1
        
        return '%dA%dB' % (bulls, cows, )

File: accepted/299-bulls-and-cows.c
/* https://leetcode.com/problems/bulls-and-cows/ */

char* getHint(char* secret, char* guess) 
{
    int nBulls = 0;
    int nCows = 0;
    
    int map[10] = { 0 };
    
    for (int i = 0; secret[i] != '\0'; i++)
        if (secret[i] == guess[i])
            nBulls++;
        else
            map[secret[i] - '0']++;
    
    for (int i = 0; secret[i] != '\0'; i++)
        secret[i] != guess[i] && map[guess[i] - '0']-- > 0 && nCows++;
    
    static char ret[32] = {0};
    sprintf(ret, "%dA%dB", nBulls, nCows);
    
    return ret;
}
